models/migrations/v98.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"strings"
+	"time"
+
+	"github.com/go-xorm/xorm"
+)
+
+func updateMigrationServiceTypes(x *xorm.Engine) error {
+	type Repository struct {
+		ID                  int64
+		OriginalServiceType int    `xorm:"index default(0)"`
+		OriginalURL         string `xorm:"VARCHAR(2048)"`
+	}
+
+	if err := x.Sync2(new(Repository)); err != nil {
+		return err
+	}
+
+	var last int
+	const batchSize = 50
+	for {
+		var results = make([]Repository, 0, batchSize)
+		err := x.Where("original_url <> '' AND original_url IS NOT NULL").OrderBy("id").Limit(batchSize, last).Find(&results)
+		if err != nil {
+			return err
+		}
+		if len(results) == 0 {
+			break
+		}
+		last += len(results)
+
+		const PlainGitService = 1 // 1 plain git service
+		const GithubService = 2   // 2 github.com
+
+		for _, res := range results {
+			u := strings.ToLower(res.OriginalURL)
+			var serviceType = PlainGitService
+			if strings.HasPrefix(u, "https://github.com") || strings.HasPrefix(u, "http://github.com") {
```

Comment:
As a good practice, it should always be `https://github.com/` rather than `https://github.com`, because currently this will match `https://github.com.mycompany.com/` (edge case, I know). Better still, you could use URL.Parse() and get the host information, so this URL passes correctly: `https://github.com:443/`. Also, lower-case normalization is desirable.